21_17_10/index.js

```diff
@@ -0,0 +1,97 @@
+const link = "https://fakestoreapi.com/products";
+const wrapper = document.getElementById("wrapper");
+const select = document.getElementById("select");
+let optionSet = new Set();
+
+function filter(card) {
+  return function sort(e) {
+    const todos = card.childNodes;
+    todos.forEach(function (todo) {
+      switch (e.target.value) {
+        case "men's clothing":
+          if (todo.getAttribute("data-category") == "men's clothing") {
+            todo.style.display = "block";
+          } else {
+            todo.style.display = "none";
+          }
+          break;
```

Comment:
для филльтрации не нужен switch.
```suggestion
if (card.getAttribute("data-category") == e.target.value) {
  card.style.display = "block";
} else {
   todo.style.display = "none";
}
```
ну и отдельную проверку надо сделать при выборе all до этого кода.